core/message_queue.go

```diff
@@ -42,6 +49,10 @@ func (mq *MessageQueue) Enqueue(msg *types.SignedMessage, stamp uint64) error {
 	mq.lk.Lock()
 	defer mq.lk.Unlock()
 
+	defer func() {
+		mqSizeGa.Set(context.TODO(), int64(len(mq.queues[msg.From])))
```

Comment:
Blocking: this is the wrong count: you need the sum over all "from" addresses. Factor out a method to do this calculation and replace the other call sites too.